ocaml/runtime/fiber.c

```diff
@@ -111,7 +114,34 @@ Caml_inline struct stack_info* alloc_for_stack (mlsize_t wosize)
   si->size = len;
   return si;
 #else
+#ifdef NATIVE_CODE
+  size_t bsize = sizeof(value) * wosize;
+  int page_size = getpagesize();
+  int num_pages = (bsize + page_size - 1) / page_size;
+  bsize = (num_pages + 2) * page_size;
+  size_t len = sizeof(struct stack_info) +
+               bsize +
+               8 /* for alignment to 16-bytes, needed for arm64 */ +
+               sizeof(struct stack_handler);
+  struct stack_info* block;
+  int memalign_res = posix_memalign((void **) &block, page_size, len);
+  if (memalign_res) {
+    return NULL;
+  }
+  block->size = len;
+  int protect = mprotect(((char*) block) + page_size, page_size, PROT_NONE);
+  if (protect) {
+    free(block);
+    return NULL;
+  }
+  return block;
+#else
+  size_t len = sizeof(struct stack_info) +
```

Comment:
We should make sure we're happy with the `Stack_threshold_ptr` computation, or ensure it isn't used if the value is wrong.

Reply:
Discussed off-line, that macro is used only
in bytecode, where the changes in this pull
request do not apply.